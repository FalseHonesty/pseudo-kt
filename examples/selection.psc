function test1(x)
    // If statements
    if x < 10 then
        print("x was less than 10")
    elseif x < 50 then
        print("x was less than 50")
    elseif x < 100 then
        print("x was less than 100")
    else
        print("x was bigger than expected. It was: " + str(x))
    endif
endfunction

test1(5)
test1(30)
test1(90)
test1(600)

function test2(x)
    // Switch statements
    switch x:
        case 10:
            print("x was 10, which is valid")
        case 50:
            print("x was 50, which is valid")
        default:
            print("x was " + str(x) + ", which is invalid")
    endswitch
endfunction

test2(2)
test2(10)
test2(50)
test2(100)
test2(-5)