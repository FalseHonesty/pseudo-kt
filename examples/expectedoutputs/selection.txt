x was less than 10
x was less than 50
x was less than 100
x was bigger than expected. It was: 600
x was 2, which is invalid
x was 10, which is valid
x was 50, which is valid
x was 100, which is invalid
x was -5, which is invalid
